query/language/reduce.feature

```diff
@@ -0,0 +1,556 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#noinspection CucumberUndefinedStep
+Feature: TypeQL Reduce Queries
+
+  Background: Open connection and create a simple extensible schema
+    Given typedb starts
+    Given connection opens with default authentication
+    Given connection has been opened
+    Given connection reset database: typedb
+    Given connection open schema transaction for database: typedb
+
+    Given typeql define
+      """
+      define
+      entity person
+        plays friendship:friend,
+        plays employment:employee,
+        owns name @card(0..),
+        owns age @card(0..),
+        owns ref @key,
+        owns email @unique @card(0..);
+      entity company
+        plays employment:employer,
+        owns name @card(0..),
+        owns ref @key;
+      relation friendship
+        relates friend @card(0..),
+        owns ref @key;
+      relation employment
+        relates employee @card(0..),
+        relates employer @card(0..),
+        owns ref @key;
+      attribute name value string;
+      attribute age value long;
+      attribute ref value long;
+      attribute email value string;
+      """
+    Given transaction commits
+
+
+  #############
+  #  reduce   #
+  #############
+
+  Scenario: 'count' returns the total number of answers
+    Given connection open write transaction for database: typedb
+    Given typeql write query
+    """
+        insert
+        $p1 isa person, has name "Klaus", has ref 0;
+        $p2 isa person, has name "Kristina", has ref 1;
+        $p3 isa person, has name "Karen", has ref 2;
+        (friend: $p1, friend: $p2) isa friendship, has ref 3;
+        """
+    Given transaction commits
+
+    Given connection open read transaction for database: typedb
+    When get answers of typeql read query
+    """
+        match
+          $x isa person;
+          $y isa name;
+          $f isa friendship;
+        """
+    Then answer size is: 9
+    When get answers of typeql read query
+    """
+        match
+          $x isa person;
+          $y isa name;
+          $f isa friendship;
+        reduce $count = count($x);
```

Comment:
Let's add another `count` case:
```
match
  $x ..
  $y ..
  $z ..
reduce $count = count($x, $y, $z)
```
which counts the number of times the triple exists without any optional components (Christoph will add the test with actual optional variables later, but we should be able to test it with multiple vars for now)